spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/secretmanager/SecretManagerConfigDataLocationResolver.java

```diff
@@ -53,14 +53,43 @@ public class SecretManagerConfigDataLocationResolver implements
    */
   private static SecretManagerServiceClient secretManagerServiceClient;
 
+  /**
+   * Checks if the property can be resolved by the Secret Manager resolver.
+   * For the check, we rely on the presence of the SecretManagerSyntaxUtils class, which is an
+   * optional dependency.
+   * Since optional dependencies may not be present at runtime, we explicitly check for its
+   * existence before resolving the property.
+   * If it's not present, it means this config resolver is not meant to be used.
+   *
+   * @return true if it contains the expected `sm@` or `sm://` prefix, false otherwise.
+   */
   @Override
   public boolean isResolvable(ConfigDataLocationResolverContext context,
       ConfigDataLocation location) {
+    boolean secretManagerSyntaxUtilsPresent = isClassPresent("com.google.cloud.spring.secretmanager.SecretManagerSyntaxUtils");
```

Comment:
@diegomarquezp A little late to this, but wonder you have this tested for GraalVM (the native test in ci may not cover this specifc scenario)? Since reflection is used here, wonder if we need to add to GraalVM hints?